pkg/source/smardex/pool_list_tracker.go

```diff
@@ -0,0 +1,160 @@
+package smardex
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+
+	"github.com/KyberNetwork/ethrpc"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/entity"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/source/pool"
+	"github.com/KyberNetwork/logger"
+)
+
+type PoolTracker struct {
+	config       *Config
+	ethrpcClient *ethrpc.Client
+}
+
+func NewPoolTracker(
+	config *Config,
+	ethrpcClient *ethrpc.Client,
+) (*PoolTracker, error) {
+	return &PoolTracker{
+		config:       config,
+		ethrpcClient: ethrpcClient,
+	}, nil
+}
+
+func (u *PoolTracker) GetNewPoolState(ctx context.Context, p entity.Pool, _ pool.GetNewPoolStateParams) (entity.Pool, error) {
+	logger.WithFields(
+		logger.Fields{"poolAddress": p.Address}).Infof(
+		"%s: Start getting new state of pool", u.config.DexID)
+
+	rpcRequest := u.ethrpcClient.NewRequest()
+	rpcRequest.SetContext(ctx)
+
+	var (
+		reserve        Reserve
+		feeToAmount    FeeToAmount
+		ethereumFees   EthereumFeeToAmount
+		fictiveReserve FictiveReserve
+		pairFee        PairFee
+		priceAverage   PriceAverage
+		totalSupply    *big.Int
+	)
+
+	rpcRequest.AddCall(&ethrpc.Call{
+		ABI:    pairABI,
+		Target: p.Address,
+		Method: pairGetReservesMethod,
+		Params: nil,
+	}, []interface{}{&reserve})
+
+	rpcRequest.AddCall(&ethrpc.Call{
+		ABI:    pairABI,
+		Target: p.Address,
+		Method: pairGetFictiveReservesMethod,
+		Params: nil,
+	}, []interface{}{&fictiveReserve})
+
+	/*
+	 * On ethereum: feesPool and feesLP are hardcode in sc
+	 * Other chains: feesPool and feesLP are gotten by rpc method
+	 * https://github.com/SmarDex-Dev/smart-contracts/commit/8d5a1e84123e07459b5dd6afbed615eacd633cc0#diff-26f384a3693d317994210fb738bb0ae5dfe485c46d867ed0b6bbf4fb7671b199
+	 */
+	if u.config.ChainID == 1 {
+		pairFee = PairFee{
+			FeesPool: big.NewInt(2),
+			FeesLP:   big.NewInt(5),
+			FeesBase: FEES_BASE_ETHEREUM,
+		}
+	} else {
+		pairFee = PairFee{
+			FeesBase: FEES_BASE_ETHEREUM,
```

Comment:
I think this should be `FEES_BASE`, as other chains have `uint256 public constant FEES_BASE = 1_000_000;`. 
```suggestion
			FeesBase: FEES_BASE,
```

Reply:
It seems correct, 10000 is outdated value, and feeLP and feePool are changed too, pr: https://github.com/SmarDex-Dev/smart-contracts/pull/7/files. I will fix it, thanks.

Reply:
Um, now I'm confused with your new changes.
I'm sorry if I'm not clear in my previous review, however, I think your previous `pairFee` with Ethereum is correct, only `pairFee` on other chains need to be updated.

For reference:
1. [Here is a Smardex Pair on Ethereum](https://etherscan.io/address/0x9fcF8F5Bd54db123470c96620441cA5C342A8Bd4#code): As defined in `SmardexLibrary.sol`, feesLP = 5, feesPool = 2, feesBase = 10000 = your previous FEES_BASE_ETHEREUM. Everything is good.
2. [Here is a Smardex Pair on Polygon](https://polygonscan.com/address/0x77476148B72ECB4E9f6A3cDDC5dd437Df1F003F3#code) which define in `SmardexLibrary.sol` `FEES_BASE = 1_000_000`, however in the old code, you set this feesBase = FEES_BASE_ETHEREUM (=10000), which I suggested to update to `FEES_BASE` (=1000000) instead.